website/docs/r/databricks_workspace.html.markdown

```diff
@@ -42,29 +42,36 @@ The following arguments are supported:
 
 * `sku` - (Required) The `sku` to use for the Databricks Workspace. Possible values are `standard`, `premium`, or `trial`. Changing this can force a new resource to be created in some circumstances.
 
-~> **NOTE** While downgrading to `trial`, the Databricks Workspace resource would be recreated.
+~> **NOTE** Downgrading to a `trial sku` from a `standard` or `premium sku` will force a new resource to be created.
 
 * `managed_resource_group_name` - (Optional) The name of the resource group where Azure should place the managed Databricks resources. Changing this forces a new resource to be created.
 
 ~> **NOTE** Azure requires that this Resource Group does not exist in this Subscription (and that the Azure API creates it) - otherwise the deployment will fail.
 
+* `customer_managed_key_enabled` - (Optional) Is the workspace enabled for customer managed key encryption? If `true` this enables the Managed Identity for the managed storage account. Possible values are `true` or `false`. Defaults to `false`. This field is only valid if the Databricks Workspace `sku` is set to `premium`. Changing this forces a new resource to be created.
```

Comment:
is there any point of this being on the resource when there is a seperate one to manage it? that cloud.. should toggle it?

Reply:
Yes, while we abstract this in being able to this functionality is a single apply via Terraform. It actually really is still a two step operation to actually implement the CMK in Databricks. This flag on the main resource tells the Databricks to create a system assigned identity behind the scenes which we will then later use to apply the key vault key access policies so the Databricks can use the key vault key to encrypt the DBFS storage.